lib/clickhousex/error.ex

```diff
@@ -665,7 +665,7 @@ defmodule Clickhousex.Error do
   defp get_code(message) do
     case Regex.scan(~r/^Code: (\d+)/i, message) do
```

Comment:
Just FYI: we can get the code from `x-clickhouse-exception-code` header.

Reply:
```console
> curl 'http://localhost:8123/?query=select' -v
*   Trying 127.0.0.1:8123...
* Connected to localhost (127.0.0.1) port 8123 (#0)
> GET /?query=select HTTP/1.1
> Host: localhost:8123
> User-Agent: curl/7.84.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 400 Bad Request
< Date: Tue, 10 Jan 2023 14:18:10 GMT
< Connection: Keep-Alive
< Content-Type: text/plain; charset=UTF-8
< X-ClickHouse-Server-Display-Name: mac3.local
< Transfer-Encoding: chunked
< X-ClickHouse-Exception-Code: 62
< Keep-Alive: timeout=10
< X-ClickHouse-Summary: {"read_rows":"0","read_bytes":"0","written_rows":"0","written_bytes":"0","total_rows_to_read":"0","result_rows":"0","result_bytes":"0"}
<
Code: 62. DB::Exception: Syntax error: failed at position 8 (end of query): . Expected one of: ALL, DISTINCT ON, DISTINCT, TOP, not empty list of expressions, list of expressions, list of elements, expression with optional alias, element of expression with optional alias, lambda expression. (SYNTAX_ERROR) (version 22.10.1.1175 (official build))
* Connection #0 to host localhost left intact
```

Reply:
I guess the error message format is unlikely to change, but the header is probably even unlikli-er to change :)

Reply:
Ah cool, I'm not sure it makes sense to work on header parsing for that with a completely new client around the corner. WYT?